internal/provider/kubernetes/controller.go

```diff
@@ -153,6 +174,8 @@ func newGatewayAPIController(ctx context.Context, mgr manager.Manager, cfg *conf
 		return fmt.Errorf("error watching resources: %w", err)
 	}
 
+	r.subscribeToResources(ctx)
```

Comment:
Same here: can we also add a comment like this to help later readers understand the change?

```
// Do not call .Subscribe() inside Goroutine since it is supposed to be called from the same
// Goroutine where Close() is called.
```